clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1499,14 +1499,16 @@ public void unsubscribe() {
         }
     }
 
-    private void resetGroupMetadata() {
-        groupMetadata.updateAndGet(
-            oldGroupMetadataOptional -> oldGroupMetadataOptional
-                .map(oldGroupMetadata -> initializeConsumerGroupMetadata(
-                    oldGroupMetadata.groupId(),
-                    oldGroupMetadata.groupInstanceId()
-                ))
-        );
+    private void maybeResetGroupMetadata() {
+        if (groupMetadata.get().isPresent()) {
```

Comment:
We do want this semantics, but I would say we don't need the check because we already get it from the `.map` call on the Optional right below (it will only map if there is a value)

Reply:
Thanks for the suggestion. I didn't notice that. Yeah, we don't need to check whether `groupMetadata` is present here.